admin/projects.go

```diff
@@ -131,7 +134,7 @@ func (s *Service) TeardownProject(ctx context.Context, p *database.Project) erro
 // UpdateProject updates a project and any impacted deployments.
 // It runs a reconcile if deployment parameters (like branch or variables) have been changed and reconcileDeployment is set.
 func (s *Service) UpdateProject(ctx context.Context, proj *database.Project, opts *database.UpdateProjectOptions) (*database.Project, error) {
-	requiresReset := (proj.Region != opts.Region) || (proj.ProdSlots != opts.ProdSlots)
+	requiresReset := (proj.Provisioner != opts.Provisioner) || (proj.ProdSlots != opts.ProdSlots) || (proj.ProdVersion != opts.ProdVersion)
```

Comment:
Updated the deployment logic to include the ability to use the provisioner to update/patch the image version without resetting the project. It's used in a new `upgrade_latest_version_projects` job, so that should take care of smoothly rolling out a new version without a lot of downtime.

In `UpdateProject` I think it's still the safest option to keep resetting the project on a version change, since we don't know if it's a version upgrade, downgrade or potentially a pre-release, so resetting the project just seems like the safest options for now and then we can refine this later on.

Reply:
Okay that makes sense. As long as regular releases don't lead to `proj.ProdVersion`?

Reply:
A release will never lead to changes to `proj.ProdVersion` by itself, the default version will be `latest` and in that case a release will trigger a deployment update, which in turn will bump the provisioned runtime to the latest release version (using the corresponding SemVar) without resetting the project.

Reply:
Perfect, that sounds great